org.mwc.cmap.plotViewer/src/org/mwc/cmap/plotViewer/actions/ExportWMF.java

```diff
@@ -78,14 +89,49 @@ public static void init(final ToolParent theParent)
 	 */
 	protected void execute()
 	{
+		IRunnableWithProgress runnable = new IRunnableWithProgress()
+		{
+			
+			@Override
+			public void run(final IProgressMonitor monitor)
+					throws InvocationTargetException, InterruptedException
+			{
+				executeInJob(monitor);
+			}
+		};
+		
+		try
+		{
+			ProgressMonitorDialog pmd = new ProgressMonitorDialog(getShell());
+			pmd.run(false, true, runnable);
+		}
+		catch (Exception e)
+		{
+			String message = e.getMessage();
+			if (message == null && e.getCause() != null)
+			{
+				message = e.getCause().getMessage();
+			}
+			MessageDialog.openError(getShell(), "Error", message);
+			CorePlugin.logError(Status.ERROR, "Tool parent missing for Write Metafile", e);
+		}
+	}
+
+	private IStatus executeInJob(IProgressMonitor monitor)
+	{
+		if (monitor.isCanceled()) {
+			return Status.CANCEL_STATUS;
+		}
 		final PlainChart theChart = getChart();
 
 		if (_theParent == null)
 		{
 			CorePlugin.logError(Status.ERROR, "Tool parent missing for Write Metafile", null);
-			return;
+			return Status.CANCEL_STATUS;
 		}
 
+		monitor.beginTask("Export to WMF", 10);
+		monitor.worked(5);

```

Comment:
we don't know how long this process will take, so I believe we can use the "unknown length" version of the progress bar.  I think it shows a "rotating pattern" that indicates something is running, but does't give an indication of %age progress.

Here's an example I found:
````
      int totalUnitsOfWork = IProgressMonitor.UNKNOWN;
      monitor.beginTask("Performing read. Please wait...", totalUnitsOfWork);
      performRead(_fileName, monitor); // This only performs the tasks
      monitor.done();
````